### ISSUE TITLE
process_batch_input_for_RNN could be shortened?

### ISSUE THREAD

**Issue by wenyong-h** [KnHuq/Dynamic-Tensorflow-Tutorial#1]
```python
    batch_input_ = tf.transpose(batch_input, perm=[2, 0, 1])
    X = tf.transpose(batch_input_)
```
should be equivalent to 
```python
    X = tf.transpose(batch_input, perm=[1, 0, 2])
```

**Comment by wenyong-h** [comment]
By the way, when i run the code under tensorflow 0.9, the loss and accuracy stay the same after every iteration.  
the result is like this:
```
Iteration: 119 Loss: 230.362 Train Accuracy: 9.42857 Test Accuracy: 12.6263
```
I also tried to run under tensorflow 0.10, the loss will change a little bit, but could almost be ignored.

**Comment by KnHuq** [comment]
@wenyong-h Your loss is not moving because you are not getting any gradient. Please check the terminal when you run the jupyter notebook. Is there any error?

**Comment by wenyong-h** [comment]
@KnHuq i'm running the python file. There was no error.
I tried using `AdamOptimizer` instead of `AdadeltaOptimizer`, then everything is right. 
this is the result of one run of the  Vanilla RNN 
```
Iteration: 119 Loss: 46.1185 Train Accuracy: 99.6429 Test Accuracy: 94.697
```

**Comment by shezan1437** [comment]
Great! Then its fine, closing the Issue!

**Comment by KnHuq** [comment]
@wenyong-h  You are right! One transpose will do the same thing!
